kart/s3_util.py

```diff
@@ -12,37 +14,82 @@
 
 # Utility functions for dealing with S3 - not yet launched.
 
+# Lots of functions in this file look like this:
+# >> @functools.lru_cache()
+# >> def _get_some_thread_unsafe_s3_resource(region, thread_hash):
```

Comment:
Maybe write a better decorator that wraps this up so you don't have to do the weird thread stuff in two versions of every function, eg `@thread_local_cache`

Reply:
Done